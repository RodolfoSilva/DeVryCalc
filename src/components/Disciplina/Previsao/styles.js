import { StyleSheet } from 'react-native'
import { colors } from '../../../styles'

export default StyleSheet.create({
  container: {
    paddingBottom: 10
  },
  value: {
    fontSize: 50,
    paddingVertical: 6
  }
})
