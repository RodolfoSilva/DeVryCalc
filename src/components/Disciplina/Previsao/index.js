import React from 'react'
import PropTypes from 'prop-types'
import Nota from '../Nota'
import styles from './styles'

const Previsao = ({ value, title }) => (
  <Nota
    value={value}
    title={title}
    style={styles.container}
    valueStyle={styles.value}
  />
)

Previsao.propTypes = {
  value: PropTypes.number,
  title: PropTypes.string.isRequired
}
export default Previsao
