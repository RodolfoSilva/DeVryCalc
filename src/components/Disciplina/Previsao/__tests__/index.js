import React from 'react'
import Previsao from '../'
import { shallow } from 'enzyme'

describe('Previsao', () => {
  it('Renderiza a previsão da nota contendo 10', () => {
    const wrapper = shallow(<Previsao value={10} title="Média" />)
    expect(wrapper).toMatchSnapshot()
  })

  it('Renderiza a previsão da nota com o placeholder', () => {
    const wrapper = shallow(<Previsao value={null} title="Média" />)
    expect(wrapper).toMatchSnapshot()
  })
})
