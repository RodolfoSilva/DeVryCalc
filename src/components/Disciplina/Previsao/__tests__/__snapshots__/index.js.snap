// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Previsao Renderiza a previsão da nota com o placeholder 1`] = `
<Nota
  style={
    Object {
      "paddingBottom": 10,
    }
  }
  title="Média"
  value={null}
  valueStyle={
    Object {
      "fontSize": 50,
      "paddingVertical": 6,
    }
  }
/>
`;

exports[`Previsao Renderiza a previsão da nota contendo 10 1`] = `
<Nota
  style={
    Object {
      "paddingBottom": 10,
    }
  }
  title="Média"
  value={10}
  valueStyle={
    Object {
      "fontSize": 50,
      "paddingVertical": 6,
    }
  }
/>
`;
